docs/proposals/gateway.md

```diff
@@ -0,0 +1,234 @@
+                fieldPath: metadata.namespace
+          - name: INSTANCE_IP
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: status.podIP
+          - name: ISTIO_META_POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          volumeMounts:
+          - name: kubeedge-certs
+            mountPath: /etc/certs
+            readOnly: true
+          - name: edgegateway-certs
+            mountPath: "/etc/kubeedge/edgegateway-certs"
+            readOnly: true
+          - name: edgegateway-ca-certs
+            mountPath: "/etc/kubeedge/edgegateway-ca-certs"
+            readOnly: true
+      volumes:
+      - name: edgegateway-certs
+        secret:
+          secretName: "istio-edgegateway-certs"
+          optional: true
+      - name: edgegateway-ca-certs
+        secret:
+          secretName: "istio-edgegateway-ca-certs"
+          optional: true
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - amd64
+                - ppc64le
+                - s390x
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 2
+            preference:
+              matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - amd64
+          - weight: 2
+            preference:
+              matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - ppc64le
+          - weight: 2
+            preference:
+              matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - s390x
+```
+
+
+
+2. User define Gateway to represent an ingress router rule to the selected EdgeGateway as well as VirtualService to represent a backend service dispatching rule. For example:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: bookinfo-gateway
+spec:
+  selector:
+    istio: edgegateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: bookinfo
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - bookinfo-gateway
+  http:
+  - match:
+    - uri:
+        exact: /productpage
+    - uri:
+        exact: /login
+    - uri:
+        exact: /logout
+    - uri:
+        prefix: /api/v1/products
+    route:
+    - destination:
+        host: productpage
+        port:
+          number: 9080
+```
+
+
+
+3. EdgeController get the definition of Gateway and VirtualService by list/watching the K8S master.

```

Comment:
nit: VirtualService is CRD installed by istio. So this would need to be optional in cases istio may not be installed - better implemented as a plugin.

Reply:
right, we should add this feature as a new module in EC